File: padroesEstruturais/padrao-de-projeto-composite/java/src/main/java/composite/Pecas/Peca.java
package composite.Pecas;

public abstract class Peca {
    private String descricao;

    public Peca setDescricacao(String descricao) {
        this.descricao = descricao;
        return this;
    }

    public String getDescricao() {
        return this.descricao;
    }
}

File: padroesEstruturais/padrao-de-projeto-composite/php/src/Pecas/Peca.php
<?php

namespace Composite\Php\Pecas;

abstract class Peca
{
    private string $descricao = '';

    public function setDescricao(string $descricao): self
    {
        $this->descricao = $descricao;
        return $this;
    }

    public function getDescricao(): string
    {
        return $this->descricao;
    }
}